### PR TITLE
[gosrc2cpg] - ORMParserPass regex issue fix

### PR DESCRIPTION
**Pull request by ankit-privado**
replaced the `[]` and `*` present in typeFullName with empty char to construct the valid regex.

### CODE REVIEW

**Review by pandurangpatil — CHANGES_REQUESTED** [review]
@ankit-privado Add the respective unit test which covers the given scenario that you are fixing.

File: src/test/scala/ai/privado/languageEngine/go/passes/orm/ORMParserTest.scala [hunk 1]
```diff
@@ -272,4 +272,123 @@ class ORMParserTest extends GoTestBase {
...
+          |  err = client.Ping(context.TODO(), nil)
+          |
+          |  if err != nil {
+          |    log.Fatal(err)
+          |  }
+          |
+          |  booksCollection := client.Database("testdb").Collection("books")
+          |  book1 := Book{"Animal Farm", "George Orwell", "0451526341", "Signet Classics", 100}
+          |  book2 := &book1
+          |  insertResult, err := booksCollection.InsertOne(context.TODO(), book2)
```
Comment (pandurangpatil) [inline src/test/scala/ai/privado/languageEngine/go/passes/orm/ORMParserTest.scala:368]: it should be `booksCollection.InsertOne(context.TODO(), &book2)`

Comment (ankit-privado) [reply src/test/scala/ai/privado/languageEngine/go/passes/orm/ORMParserTest.scala]: fixed now. @pandurangpatil

**Review by pandurangpatil — APPROVED** [review]
Looks good

**Pull request merged**